app/helpers/description.py

```diff
@@ -0,0 +1,55 @@
+import json
+import logging
+import os
+
+from flask import abort
+
+from app.settings import DESCRIPTION_FOLDER
+
+logger = logging.getLogger(__name__)
+
+
+def get_icon_set_description(icon_set=''):
+    '''
+    Return json containing the description in all available languages for all icons of the
+    provided icon set
+    '''
+    path = find_descripton_file(icon_set)
+    if not os.path.isfile(path):
+        return None
+
+    with open(path, encoding='utf-8') as f:
+        icon_set_descriptions = json.load(f)
+
+    return icon_set_descriptions
+
+
+def get_icon_description(icon_name='', icon_set=''):
+    '''
+    Return json containing the description in all available languages for the specified icon of the
+    provided icon set
+    '''
+    path = find_descripton_file(icon_set)
+    if not os.path.isfile(path):
+        return None
+
+    with open(path, encoding='utf-8') as f:
+        df = json.load(f)
+
+    try:
+        icon_description = df[icon_name]
+    except KeyError:
+        logger.error("Description for icon not found: %s", icon_name)
+        abort(404, "Description for icon not found")
+
+    return icon_description
+
+
+def find_descripton_file(icon_set):
```

Comment:
could the check for existence of the description file not be done directkly also in this function instead of a separate?
What's the reason for doing that separately?

Reply:
Ah, sorry, I see. It is a valid case that no description file exists. So it doesn't make sense to check for existence here and raise an error if it doesn't exist